core/src/main/java/com/scalar/db/transaction/common/AbstractTwoPhaseCommitTransactionManager.java

```diff
@@ -188,8 +195,148 @@ public void rollback() throws RollbackException {
       }
     }
 
+    @Override
+    public TwoPhaseCommitTransaction getOriginalTransaction() {
+      if (transaction instanceof WrappedTwoPhaseCommitTransaction) {
+        return ((WrappedTwoPhaseCommitTransaction) transaction).getOriginalTransaction();
+      }
+      return transaction;
+    }
+  }
+
+  @VisibleForTesting
+  static class TransactionStateManagement extends AbstractTwoPhaseCommitTransaction
```

Comment:
This class extends (and implements) `XxxxxxCommitTransaction`, but the class name sounds a bit different layer.

Reply:
Something like StateManagedTransaction or StateManageableTransaction?
(both don't sound very cool, though...)

BTW, it might not make sense, but managing active or not is part of state management, so can we unify `ActiveTransaction` and `TransactionStateManagement`?

Reply:
Honestly, I didn't think it was a good name for this class, but I didn't have any other idea. 

> BTW, it might not make sense, but managing active or not is part of state management, so can we unify ActiveTransaction and TransactionStateManagement?

Maybe, I feel it's better to have separate classes for those. I will rename it to `StateManagedTransaction`. Thank you. BTW, it's a private inner class so we can rename it whenever we want in the future.

Reply:
Renamed it in 76c9ae03b1695099a82c97cc7ac84b978378eda2. Thanks.